File: hackathon/security-copilot/frontend/src/types/content.ts
export enum ContentType {
  ExecutiveBrief = 'executive_brief',
  CustomerAdvisory = 'customer_advisory',
  TechnicalAnalysis = 'technical_analysis',
  Newsletter = 'newsletter',
  SocialMedia = 'social_media',
}

export interface GeneratedContent {
  id: number;
  article_id: number;
  content_type: ContentType;
  title: string | null;
  content: string;
  is_approved: boolean;
  created_at: string;
  updated_at: string;
}

export interface GeneratedContentUpdate {
  title?: string;
  content?: string;
}

export interface PublishedContent {
  id: number;
  generated_content_id: number;
  article_id: number;
  published_by: number | null;
  platform: string;
  status: string;
  published_at: string;
}

export interface PublishHistoryParams {
  skip?: number;
  limit?: number;
}

export interface PublishHistoryResponse {
  items: PublishedContent[];
  total: number;
}

File: hackathon/security-copilot/backend/app/schemas/content.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class GeneratedContentResponse(BaseModel):
    id: int
    article_id: int
    content_type: str
    title: Optional[str] = None
    content: str
    image_url: Optional[str] = None
    is_approved: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class GeneratedContentUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    is_approved: Optional[bool] = None


class PublishedContentResponse(BaseModel):
    id: int
    generated_content_id: int
    article_id: int
    published_by: Optional[int] = None
    platform: str
    status: str
    response_message: Optional[str] = None
    published_at: datetime

    model_config = ConfigDict(from_attributes=True)


class PublishedContentListResponse(BaseModel):
    items: list[PublishedContentResponse]
    total: int


class GeneratedContentListResponse(BaseModel):
    items: list[GeneratedContentResponse]
    total: int
    skip: int
    limit: int
